### PR TITLE
Signal profiler & proto changes

### PR DESCRIPTION
**Pull request by xxuejie**
This PR adds a series of changes:

* It adds [ckb-vm-signal-profiler](https://github.com/cryptape/ckb-vm-signal-profiler) to the monorepo.
* Since signal profiler and pprof converter both uses pprof proto file, a single crate is added to handle proto compilation.
* [Vendored protoc](https://crates.io/crates/protoc-bin-vendored) is now only used on Linux. We will rely on homebrew to provide `protoc` on macOS, since vendored protoc does not provide binary for Apple Silicon.

### CODE REVIEW

**Comment by mohanson** [comment]
Failed to run `cargo run --example simple` with outputs:

```
error[E0061]: this function takes 1 argument but 2 arguments were supplied
   --> ckb-vm-signal-profiler/examples/simple.rs:61:32
    |
61  |     let mut machine = Box::pin(AsmMachine::new(core, None));
    |                                ^^^^^^^^^^^^^^^       ---- argument of type `Option<_>` unexpected
    |
note: associated function defined here
   --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/ckb-vm-0.22.2/src/machine/asm/mod.rs:430:12
    |
430 |     pub fn new(machine: DefaultMachine<'a, Box<AsmCoreMachine>>) -> Self {
    |            ^^^
help: remove the extra argument
    |
61  |     let mut machine = Box::pin(AsmMachine::new(core));
    |                                               ~~~~~~

For more information about this error, try `rustc --explain E0061`.
error: could not compile `ckb-vm-signal-profiler` due to previous error
```

**Comment by xxuejie** [comment]
> Failed to run `cargo run --example simple` with outputs:
> 
> ```
> error[E0061]: this function takes 1 argument but 2 arguments were supplied
>    --> ckb-vm-signal-profiler/examples/simple.rs:61:32
>     |
> 61  |     let mut machine = Box::pin(AsmMachine::new(core, None));
>     |                                ^^^^^^^^^^^^^^^       ---- argument of type `Option<_>` unexpected
>     |
> note: associated function defined here
>    --> /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/ckb-vm-0.22.2/src/machine/asm/mod.rs:430:12
>     |
> 430 |     pub fn new(machine: DefaultMachine<'a, Box<AsmCoreMachine>>) -> Self {
>     |            ^^^
> help: remove the extra argument
>     |
> 61  |     let mut machine = Box::pin(AsmMachine::new(core));
>     |                                               ~~~~~~
> 
> For more information about this error, try `rustc --explain E0061`.
> error: could not compile `ckb-vm-signal-profiler` due to previous error
> ```

Already fixed.

File: ckb-vm-signal-profiler/examples/simple.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+fn main() {
+    env_logger::init();
+
+    let args: Vec<String> = env::args().skip(1).collect();
+    let code = std::fs::read(args[0].clone()).unwrap().into();
+    let args: Vec<Bytes> = args.into_iter().map(|a| a.into()).collect();
+
+    let asm_core = AsmCoreMachine::new(ISA_IMC, VERSION1, u64::max_value());
```
Comment (mohanson) [inline ckb-vm-signal-profiler/examples/simple.rs]: It would be nicer to add the `ISA_B` and `ISA_MOP`. I just failed because of using a toolchain with B.

Comment (xxuejie) [reply ckb-vm-signal-profiler/examples/simple.rs]: Oops I forgot about this, already fixed.

**Review by mohanson — APPROVED** [review]
An amazing tool!

**Pull request merged**